internal/db/sqltest/tests/iam/grant_scope_validation.sql

```diff
@@ -27,113 +30,239 @@ insert into iam_role (public_id, scope_id) values
+    values
+        ('r_111111111a', 'o_1111111111');
+select throws_like(
+    'insert_bare_proj_fail_org',
+    'invalid to set a grant scope ID to non-same scope_id_or_special when role scope type is project'
+);
+prepare insert_bare_proj_wrong_scope_fail_org as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_1111111111', 'p_222222222a');
+select throws_like(
+    'insert_bare_proj_wrong_scope_fail_org',
+    'grant scope id is not a child project of the role''s org scope'
+);
+
+-- Reset grant scopes
+delete from iam_role_grant_scope;
+
+-- Case 3: "children" is only allowed in global/org
+prepare insert_children_global as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_globeglobe', 'children');
+select lives_ok('insert_children_global');
+prepare insert_children_org as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_1111111111', 'children');
+select lives_ok('insert_children_org');
+prepare insert_children_fail_proj as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_111111111a', 'children');
+select throws_like(
+    'insert_children_fail_proj',
+    'invalid to set a grant scope ID to non-same scope_id_or_special when role scope type is project'
+);
+
+-- Reset grant scopes
+delete from iam_role_grant_scope;
+
+-- Case 4: "descendants" is only allowed in global
+prepare insert_descendants_global as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_globeglobe', 'descendants');
+select lives_ok('insert_descendants_global');
+prepare insert_descendants_fail_org as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_1111111111', 'descendants');
+select throws_like(
+    'insert_descendants_fail_org',
+    'invalid to specify "descendants" as a grant scope when the role''s scope ID is not "global"'
+);
+prepare insert_descendants_fail_proj as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_111111111a', 'descendants');
+select throws_like(
+    'insert_descendants_fail_proj',
+    'invalid to set a grant scope ID to non-same scope_id_or_special when role scope type is project'
+);
+
+-- Reset grant scopes
+delete from iam_role_grant_scope;
+
+-- Case 5: "descendants" and "children" are mutually exclusive
+prepare insert_descendants__and_children_fail_global as
+    insert into iam_role_grant_scope
+        (role_id,        scope_id_or_special)
+    values
+        ('r_globeglobe', 'descendants'),
+        ('r_globeglobe', 'children');
+select throws_like(
+    'insert_descendants__and_children_fail_global',
+    'invalid to specify both "children" and "descendants"as a grant scope'
```

Comment:
It looks like there is a [space missing in the error message](https://github.com/hashicorp/boundary/blob/077f84eb3985d2da1652272823c1ee9b1894391e/internal/db/schema/migrations/oss/postgres/83/01_iam_role_grant_scope.up.sql#L124-L132). Should probably update that and then this assertion can be updated.

```suggestion
    'invalid to specify both "children" and "descendants" as a grant scope'
```

Reply:
Fixed